topo_processor/util/s3.py

```diff
@@ -1,8 +1,32 @@
+import boto3
+from linz_logger import get_log
+
+
 def bucket_name_from_path(path: str) -> str:
     path_parts = path.replace("s3://", "").split("/")
     return path_parts.pop(0)
 
 
+def bucket_name_from_stack(stack_name: str) -> str:
+    get_log().debug("stack_name", stack_name=stack_name)
+    session = boto3.Session()
+    cloudformation = session.resource("cloudformation")
+    stack = cloudformation.Stack(stack_name)
+
+    temp_bucket: str = ""
+
+    try:
+        for output in stack.outputs:
+            if output["OutputKey"] == "TempBucketName":
+                get_log().debug("bucket_name", bucket_name=output["OutputValue"])
+                temp_bucket = output["OutputValue"]
+    except Exception as e:
+        get_log().error("bucket_name_not_found", stackName=stack_name, error=e)
```

Comment:
This error will be logged if an exception is encountered - and it could be one of the reason that the bucket is not found with that name. But what about if there is no exception raised and the bucket is also not found: i.e. there is no `OutputKey` with that name, so `temp_bucket` is still `== ""`?